milestones/api.py

```diff
@@ -163,7 +163,10 @@ def get_course_milestones(course_key, relationship=None):
     _validate_course_key(course_key)
 
     if relationship is not None:
-        _validate_milestone_relationship_type(relationship)
+        try:
+            _validate_milestone_relationship_type(relationship)
+        except exceptions.InvalidMilestoneRelationshipTypeException:
+            return []
```

Comment:
Ah, right -- sorry, it's been a while since I've looked at edx-milestones :)  It appears that if we drop the relationship type validation then the system will return an InvalidMilestoneRelationshipType exception from the data layer -- ie, we get the same effect.  So yeah, I agree that in this case we probably do not need the validation in the api layer.